File: cookbooks/python/hn-time-capsule/schema.py
"""Structured output schema for HN Time Capsule analysis."""

from pydantic import BaseModel, Field


class Award(BaseModel):
    user: str = Field(description="HN username")
    reason: str = Field(description="Why they were right/wrong in hindsight")


class Analysis(BaseModel):
    title: str = Field(description="Article title")
    summary: str = Field(description="Brief summary of article and discussion")
    what_happened: str = Field(description="What actually happened to this topic/company/technology")
    most_prescient: Award = Field(description="Commenter who best predicted the future")
    most_wrong: Award = Field(description="Commenter who was most wrong")
    grades: dict[str, str] = Field(description="HN username → letter grade (A+ to F)")
    score: int = Field(description="0-10 how interesting this retrospective is")

File: cookbooks/typescript/hn-time-capsule/schema.ts
/**
 * Structured output schema for HN Time Capsule analysis.
 */

import { z } from "zod";

const AwardSchema = z.object({
    user: z.string().describe("HN username"),
    reason: z.string().describe("Why they were right/wrong in hindsight"),
});

export const AnalysisSchema = z.object({
    title: z.string().describe("Article title"),
    summary: z.string().describe("Brief summary of article and discussion"),
    what_happened: z.string().describe("What actually happened to this topic/company/technology"),
    most_prescient: AwardSchema.describe("Commenter who best predicted the future"),
    most_wrong: AwardSchema.describe("Commenter who was most wrong"),
    grades: z.record(z.string()).describe("HN username → letter grade (A+ to F)"),
    score: z.number().describe("0-10 how interesting this retrospective is"),
});
